subprojects/problems/src/main/java/org/gradle/api/problems/internal/DefaultProblemBuilder.java

```diff
@@ -244,6 +242,10 @@ public RuntimeException throwError(RuntimeException exception, Problem problem)
     }
 
     private void report(Problem problem) {
-        buildOperationProgressEventEmitter.emitNowIfCurrent(problem);
+        problemsService.reportAsProgressEvent(problem);
+    }
```

Comment:
🤔 Maybe this should be removed, and we should rely on `ReportableProblem` to make the reporting.

Reply:
With the correct approach there are reportable and the others are not. You can create `ReportableProbem` with Problems.createProblemBuilder(), but not within `throwing()` nor in `rethrowing()`